### ISSUE TITLE
Thoughts about project 

### ISSUE THREAD

**Issue by denysdovhan** [everylint/everylint#21]
As I said in #19, I was thinking about such tool for a few months, so I have some suggestions about how the project might evolve:

### Principles

* *Have reasonable defaults* — by default everylint should have only _reasonable_ defaults. It should not enforce styling rules (they might be different from project to project), but only the most important ones, like `no-endef`.
* *Work out of the box* — since it should include reasonable defaults, it also should work out of the box, without any additional configuration. Just run `everylint` from the console, and you'll get a comprehensive report about all of the files.
* *Be configurable* — despite the fact, that everylint should work out of the box with reasonable defaults, it should also be configurable. Users want to disable, enable or configure additional or existing rules easily.

### Linting by _type of file_ instead of linting by _linter_

The current implementation uses _liters_ for linting. However, most of the users might want to switch and use different linters (`htmllint` vs. `pug-lint`, `remark-lint` vs. `markdownlint`). Instead, I'd suggest using abstraction, like `file` with its `type`, like:

* `scripts` — for `js`, `ts`, `coffee`, etc.
* `styles` — for `css`, `scss`, `styl`, `pcss`, etc.
* `template` — for `html`, `pug`, `.ejs`, `haml`, etc.
* `assets` — for images, other assets.

Depending on `type`, we can run desirable linters:

* `scripts` — `eslint` or `tslint` or `jshint` or other.
* `styles` — `stylelint` or other.
* `template` — `htmllint` or `pug-lint` or other.
* `assets` — `size-limit` or other.

We can also group report by type of file.

### Using `VFile` as a file abstraction

[`vfile`](https://github.com/vfile) is an excellent abstraction for virtual files. To reduce reading of files, we can globe them by extensions, read them once and then pass their content to different linters. `vfile` also gives us the ability to attach report messages to the file.

I'll gladly help to make it project better.

What do you think?

**Comment by Zmoki** [comment]
@denysdovhan I agree with that principles. I need a little time to think it over and study the `vfile` project. I'll be back soon with a more detailed answer.
